lib/tool_shed/util/repository_util.py

```diff
@@ -170,6 +170,81 @@ def create_repository( app, name, type, description, long_description, user_id,
     message = "Repository <b>%s</b> has been created." % escape( str( repository.name ) )
     return repository, message
 
+def update_repository( app, trans, id, **kwds):
+    """Update an existing ToolShed repository"""
+    sa_session = app.model.context.current
+    # Add the repository record to the database.
+    repository = sa_session.query( app.model.Repository ).get( app.security.decode_id( id ) )
```

Comment:
Yeah but previously this was not part of an API so it shouldn't have ever been called with inexistent IDs (still improvable design though). See below.